Compilation fails on ESPHome 2026.8.0

its bean a while that esphome team made a breaking change in ble :)

Error log snippet:
error: 'const class esphome::ble_device_base::ESPBTUUID' has no member named 'to_string'; did you mean 'to_str'?

in `eqiva_listener.cpp`:

I tried with this config bit it still fails for me - it has been some time.

```
external_components:
  - source: github://digaus/esphome-components-eqiva@main
    refresh: 1s
    components: [ eqiva_ble, eqiva_key_ble ]

```
Log:

```
INFO Updating https://github.com/digaus/esphome-components-eqiva.git@main
...
FAILED: esp-idf/src/CMakeFiles/__idf_src.dir/esphome/components/eqiva_ble/eqiva_listener.cpp.obj
ccache /data/cache/idf/tools/xtensa-esp-elf/esp-14.2.0_20260121/xtensa-esp-elf/bin/xtensa-esp32-elf-g++ @esp-idf/src/CMakeFiles/__idf_src.dir/esphome/components/eqiva_ble/eqiva_listener.cpp.obj.rsp -MD -MT esp-idf/src/CMakeFiles/__idf_src.dir/esphome/components/eqiva_ble/eqiva_listener.cpp.obj -MF esp-idf/src/CMakeFiles/__idf_src.dir/esphome/components/eqiva_ble/eqiva_listener.cpp.obj.d -o esp-idf/src/CMakeFiles/__idf_src.dir/esphome/components/eqiva_ble/eqiva_listener.cpp.obj -c /data/build/keyble-wohnungstuer/src/esphome/components/eqiva_ble/eqiva_listener.cpp
In file included from ../src/esphome/core/gpio.h:6,
                 from ../src/esphome/core/hal.h:5,
                 from ../src/esphome/core/component.h:9,
                 from ../src/esphome/components/eqiva_ble/eqiva_listener.h:5,
                 from ../src/esphome/components/eqiva_ble/eqiva_listener.cpp:1:
../src/esphome/components/eqiva_ble/eqiva_listener.cpp: In member function 'virtual bool esphome::eqiva_ble::EqivaListener::parse_device(const esphome::esp32_ble_tracker::ESPBTDevice&)':
../src/esphome/components/eqiva_ble/eqiva_listener.cpp:15:100: warning: 'std::string esphome::ble_device_base::ESPBTDevice::address_str() const' is deprecated: Use address_str_to() instead. Removed in 2027.2.0. [-Wdeprecated-declarations]
   15 |             ESP_LOGI(TAG, "Found Eqiva device (MAC: %s) (UUID): %s  (Name): %s", device.address_str().c_str(), it.uuid.to_str().c_str(), device.get_name().c_str());
      |                                                                                  ~~~~~~~~~~~~~~~~~~^~
../src/esphome/core/log.h:119:99: note: in definition of macro 'esph_log_i'
  119 |   ::esphome::esp_log_printf_(ESPHOME_LOG_LEVEL_INFO, tag, __LINE__, ESPHOME_LOG_FORMAT(format), ##__VA_ARGS__)
      |                                                                                                   ^~~~~~~~~~~
../src/esphome/components/eqiva_ble/eqiva_listener.cpp:15:13: note: in expansion of macro 'ESP_LOGI'
   15 |             ESP_LOGI(TAG, "Found Eqiva device (MAC: %s) (UUID): %s  (Name): %s", device.address_str().c_str(), it.uuid.to_str().c_str(), device.get_name().c_str());
      |             ^~~~~~~~
In file included from ../src/esphome/components/esp32_ble_tracker/esp32_ble_tracker.h:22,
                 from ../src/esphome/components/eqiva_ble/eqiva_listener.h:6:
../src/esphome/components/ble_device_base/ble_device.h:189:15: note: declared here
  189 |   std::string address_str() const;
      |               ^~~~~~~~~~~
../src/esphome/components/eqiva_ble/eqiva_listener.cpp:15:126: error: no matching function for call to 'esphome::ble_device_base::ESPBTUUID::to_str() const'
   15 |             ESP_LOGI(TAG, "Found Eqiva device (MAC: %s) (UUID): %s  (Name): %s", device.address_str().c_str(), it.uuid.to_str().c_str(), device.get_name().c_str());
      |                                                                                                                ~~~~~~~~~~~~~~^~
../src/esphome/core/log.h:119:99: note: in definition of macro 'esph_log_i'
  119 |   ::esphome::esp_log_printf_(ESPHOME_LOG_LEVEL_INFO, tag, __LINE__, ESPHOME_LOG_FORMAT(format), ##__VA_ARGS__)
      |                                                                                                   ^~~~~~~~~~~
../src/esphome/components/eqiva_ble/eqiva_listener.cpp:15:13: note: in expansion of macro 'ESP_LOGI'
   15 |             ESP_LOGI(TAG, "Found Eqiva device (MAC: %s) (UUID): %s  (Name): %s", device.address_str().c_str(), it.uuid.to_str().c_str(), device.get_name().c_str());
      |             ^~~~~~~~
../src/esphome/components/ble_device_base/ble_device.h:91:15: note: candidate: 'const char* esphome::ble_device_base::ESPBTUUID::to_str(char*) const'
   91 |   const char *to_str(char *buf) const;
      |               ^~~~~~
../src/esphome/components/ble_device_base/ble_device.h:91:15: note:   candidate expects 1 argument, 0 provided
../src/esphome/components/ble_device_base/ble_device.h:93:15: note: candidate: 'const char* esphome::ble_device_base::ESPBTUUID::to_str(std::span<char, 37>) const'
   93 |   const char *to_str(std::span<char, UUID_STR_LEN> output) const { return this->to_str(output.data()); }
      |               ^~~~~~
../src/esphome/components/ble_device_base/ble_device.h:93:15: note:   candidate expects 1 argument, 0 provided
[14/78] Building CXX object esp-idf/src/CMakeFiles/__idf_src.dir/esphome/components/network/util.cpp.obj
ninja: build stopped: subcommand failed.
ninja failed with exit code 1, output of the command is in the /data/build/keyble-wohnungstuer/build/log/idf_py_stderr_output_116097 and /data/build/keyble-wohnungstuer/build/log/idf_py_stdout_output_116097
```

--- a/components/eqiva_ble/eqiva_listener.cpp
+++ b/components/eqiva_ble/eqiva_listener.cpp
@@ -12,7 +12,8 @@ static const char *const TAG = "eqiva_ble";
 bool EqivaListener::parse_device(const esp32_ble_tracker::ESPBTDevice &device) {
     for (auto &it : device.get_manufacturer_datas()) {
         if (it.uuid == esp32_ble_tracker::ESPBTUUID::from_uint16(0x1A00)) {
-            ESP_LOGI(TAG, "Found Eqiva device (MAC: %s) (UUID): %s  (Name): %s", device.address_str().c_str(), it.uuid.to_string().c_str(), device.get_name().c_str());
+            char uuid_buf[37];  // ESPBTUUID::UUID_STR_LEN
+            ESP_LOGI(TAG, "Found Eqiva device (MAC: %s) (UUID): %s  (Name): %s", device.address_str().c_str(), it.uuid.to_str(uuid_buf), device.get_name().c_str());
             return true;
         }
     }

